packages/web/src/js/Collapse.ts

```diff
@@ -59,109 +61,149 @@ class Collapse extends BaseComponent {
     return `${this.NAME}`;
   }
 
-  onResize() {
-    this.state.width = window.innerWidth;
-    this.updateTriggerElementHandler();
-    this.updateCollapsibleElementHandler();
+  hideParentSiblings(trigger: HTMLElement) {
+    if (this.meta.parent) {
+      const parentElement = SelectorEngine.findOne(this.meta.parent);
+      const children = parentElement?.children;
+
+      if (!children) {
+        return;
+      }
+
+      for (const item of children) {
+        const itemTrigger = SelectorEngine.findOne(`[${NAME_DATA_TOGGLE}="${NAME}"]`, item);
+        const instance = Collapse.getInstance(itemTrigger);
+
+        if (instance?.state?.open && itemTrigger !== trigger) {
+          instance?.hide();
+        }
+      }
+    }
   }
 
-  appendNodeToParentHandler() {
-    if (this.state.collapsed) {
+  async appendNodeToParent() {
+    if (this.state.open) {
       const contentEl = this.target?.children[0];
       const innerHtml = contentEl ? contentEl?.innerHTML : this.target?.innerHTML;
       const originalHtml = this.parent?.innerHTML;
       if (this.parent) {
         this.parent.innerHTML = `${originalHtml}${innerHtml}`;
       }
-      setTimeout(() => this.target?.remove(), 100);
+
+      await this.target?.remove();
     }
   }
 
-  adjustCollapsibleElementHeightHandler(collapsed: boolean = this.state.collapsed) {
+  adjustCollapsibleChildrenHeight(zeroHeight?: boolean) {
     if (this.target) {
```

Comment:
In both cases where you call this, there is also a check for `this.target`. It is necessary to recheck the target here again? Maybe it will be better to pass the target as an argument.

Reply:
Not true, it is only one case when i call this method in `if (this.target)` scope and it is scope for whole function.